Tensile/TensileCreateLibrary.py

```diff
@@ -103,31 +103,109 @@ def processKernelSource(kernel, kernelWriterSource, kernelWriterAssembly) -> Pro
     return (err, src, header, kernelName, filename)
 
 
-def getAssemblyCodeObjectFiles(kernels, kernelWriterAssembly, outputPath, removeTemporaries):
-    destDir = ensurePath(os.path.join(outputPath, "library"))
-    asmDir = kernelWriterAssembly.getAssemblyDirectory()
-    assemblyKernels = list([k for k in kernels if k["KernelLanguage"] == "Assembly"])
+def linkIntoCodeObject(
+    objFiles: List[str], coPathDest: Union[Path, str], kernelWriterAssembly: KernelWriterAssembly
+):
+    """Links object files into a code object file.
+
+    Args:
+        objectFiles: A list of object files to be linked.
+        coPathDest: The destination path for the code object file.
+        kernelWriterAssembly: An instance of KernelWriterAssembly to get link arguments.
+
+    Raises:
+        RuntimeError: If linker invocation fails.
+    """
+    args = []
+    if os.name == "nt":
+        # On Windows, the objectFiles list command line (including spaces)
+        # exceeds the limit of 8191 characters, so using response file
+
+        responseFile = os.path.join("/tmp", "clangArgs.txt")
+        with open(responseFile, "wt") as file:
+            file.write(" ".join(objFiles))
+            file.flush()
+
+        args = kernelWriterAssembly.getLinkCodeObjectArgs(["@clangArgs.txt"], str(coPathDest))
+    else:
+        args = kernelWriterAssembly.getLinkCodeObjectArgs(objFiles, str(coPathDest))
+
+    tPrint(2, "Linking objects into co files: " + " ".join(args))
+    try:
+        out = subprocess.check_output(args, stderr=subprocess.STDOUT)
+        tPrint(2, f"Output: {out}")
+    except subprocess.CalledProcessError as err:
+        raise RuntimeError(
+            f"Error linking object files in to code object: {err.output}\nFailed command: {' '.join(args)}"
+        )
+
+
+def compressCodeObject(
+    coPathSrc: Union[Path, str], coPathDest: Union[Path, str], gfx: str, bundler: str
+):
+    """Compresses a code object file using the provided bundler.
+
+    Args:
+        coPathSrc: The source path of the code object file to be compressed.
+        coPathDest: The destination path for the compressed code object file.
+        gfx: The target GPU architecture.
+        bundler: The path to the Clang Offload Bundler executable.
+
+    Raises:
+        RuntimeError: If compressing the code object file fails.
+    """
+    args = [
+        bundler,
+        "--compress",
+        "--type=o",
+        "--bundle-align=4096",
```

Comment:
Where did we get 4096 and should it be fixed?

Reply:
This parameter was recommended to me by the compiler team. @yxsamliu how does this parameter affect the output? Is it needed in this case?